framework/src/variables/MooseVariableData.C

```diff
@@ -192,6 +192,17 @@ MooseVariableData<OutputType>::MooseVariableData(const MooseVariableFE<OutputTyp
   _grad_phi_face = &_grad_phi_face_assembly_method(_assembly, _fe_type);
 }
 
+template <typename OutputType>
+unsigned int
+MooseVariableData<OutputType>::needSolutionState() const
+{
+  if (_need_u_older || _need_grad_older || _need_second_older || _need_dof_values_older)
+    return 2;
```

Comment:
Hmm, just looking at the language in this method, maybe I'm more on the @fdkong side of a `MooseEnum`. Unless we truly do have a higher solution state somewhere that doesn't have *any* associated `older[er]*` language

Reply:
In an ideal world, I would actually have the abstract state design within variable data as well. What do you think about that?

Reply:
I'm in favor. I would expect it to decrease # LOC